druid-shell/src/util.rs

```diff
@@ -43,6 +44,22 @@ pub fn assert_main_thread() {
     }
 }
 
+/// Assert that the current thread is the registered main thread or main thread is not claimed.
+///
+/// # Panics
+///
+/// Panics when called from a non-main thread and main thread is claimed.
+pub fn assert_main_thread_or_main_unclaimed() {
```

Comment:
does this need to be public?

Reply:
all other functions in util are also public but util is not public. I will make all these `pub(crate)`